missioncontrol_client/__init__.py

```diff
@@ -145,6 +239,75 @@ def get_pass_task_stack(self, uuid, **kwargs):
             json=kwargs
         )
 
+    def get_files(self, **kwargs):
+        return self.getj(
+            f'/api/v0/files/',
+            params=kwargs
+        )
+
+    def get_file(self, cid):
+        return self.getj(
+            f'/api/v0/files/{cid}/'
+        )
+
+    def download_file(self, cid):
+        return self.get(
+            f'/api/v0/files/{cid}/data/'
+        )
+
+    def upload_file(self, path, what, where=None, start=None, end=None,
+                    work_id=None, content_type=None):
+        if start is None:
+            start = UTC("now").iso
+
+        if where is None:
+            where = socket.gethostname()
+
+        f = GzippedFileStream.from_filename(
+            path,
+            what=what,
+            where=where,
+            start=start,
+            work_id=work_id
+        )
+
+
+        cid = f.metadata["hash"]
+
+        fleetmeta = {
```

Comment:
Fleet specific?

Reply:
I'm inheriting from planetlabs/datalake File class, but we've changed our metadata structure, so this converts to the "fleet" version of metadata.

Ideally we'd fork/enhance this library with `metadata 2.0` and distribute it, but using it as is provides a lot of value without much effort in the near term.

Reply:
Ah right.
At this point I don't actually see what this File class adds to us? Can you explain a bit?

Reply:
Yeah - I'm trying to leverage as much of the datalake infrastructure as possible, as it includes lessons learned and features developed over several years.
For example, datalake files have a `tar bundle` format, and an inotify-based auto-uploaded service. If we can re-use a lot of this work, we won't have to rewrite it from scratch.

Reply:
Except currently we're just doing a POST directly anyway?
I'm wondering what specifically *this* PR users from the `File` class of `datalake`

Maybe the answer is 'nothing yet'?

Reply:
I think the mismatch is that I was overriding the methods used in this diff to get streaming gzip working.
I've since moved that into a datalake fork. I think it's worth factoring out the metadata and tooling as it's a bit more complex than just normal REST api stuff.
Ideally this library would be very lean and not doing too much magic.